lib/discovery/updater_test.go

```diff
@@ -143,27 +143,29 @@ func TestUpdater_OnEvent_AdditionAndModifyAndDelete_HTTP(t *testing.T) {
 				{
 					Autogenerated: true,
 					HostMatcher:   "s2.external2.host.com",
-					PortMatcher:   222,
+					PortMatcher:   80,
```

Comment:
i gave up on this whole file. We should think about refactoring this somehow. It is super hard to follow or figure out what is going ion

Reply:
hm, I separated these into two different files for grpc and http...but not sure how we can make it simpler or more clean